server/http/src/service.rs

```diff
@@ -272,9 +272,41 @@ where
         let query = GraphQLRequest::new(body, schema, network).compat().await;
 
         let result = match query {
-            Ok(query) => graph::spawn_blocking_allow_panic(service.graphql_runner.run_query(query))
-                .await
-                .unwrap(),
+            Ok(query) => {
+                // Not great that we serialize the query here only for the panic case.
+                let query_text = query.query_text();
+                let variables_text = query.variables_text();
+
+                let result =
+                    graph::spawn_blocking_allow_panic(service.graphql_runner.run_query(query))
+                        .await;
+
+                match result {
+                    Ok(res) => res,
+
+                    // `JoinError` means a panic.
```

Comment:
This comment seems like it was left over from a previous incarnation of the code

Reply:
The type of `e` here is `JoinError`. I'll make this `Err(JoinError)` in the comment for clarity.